src/main/java/com/jcraft/jsch/JSch.java

```diff
@@ -42,7 +43,12 @@
 
   static Hashtable<String, String> config=new Hashtable<>();
   static{
+    fillConfig(config, JavaVersion.getVersion());
+  }
+
+  static void fillConfig(Map<String, String> config, int javaVersion) {
```

Comment:
What's the purpose of the `javaVersion` argument? It appears to be unused?

Reply:
It allows to test the different behavior in dependence of different java versions without the requirement to actually use that particular java version when executing the test

Reply:
What's missing is a checkFillConfiig in JschTest but that's a big TODO I haven't had the time to attack

Reply:
... and the actual use of the parameter was missing as well. Not sure what happened here, I've definitely used it when doing the implementation.

Reply:
I'm still not sure I understand why we need a `javaVersion` argument instead of simply using `JavaVersion.getVersion()`?
The value returned by `JavaVersion.getVersion()` doesn't dynamically change at runtime.

Reply:
I've committed JschTest showing the use of this parameter as part of a unit test. The fact that JavaVersion.getVersion doesn't change at runtime is the actual problem if you want to test behavior that depends on this value